File: October 2024/Pairs with difference k( 30 oct ).py
class Solution:
    def countPairsWithDiffK(self, arr, k):
        unique_elements = set(arr)
        count = 0
        
        # Check for each element if element + k exists in the set
        for x in unique_elements:
            if x + k in unique_elements:
                count += 1
        
        return count

File: October 2024/Pairs with difference k(30 oct).java
class Solution {
    int countPairsWithDiffK(int[] arr, int k) {
        // code here
        int ans = 0;
        Map<Integer,Integer> mm = new HashMap<>();
        for(int x:arr)mm.put(x,mm.getOrDefault(x,0)+1);
        for(int x:arr)ans+=mm.getOrDefault(x+k,0);
        return ans;
    }

}
